ports/raspberrypi/common-hal/mdns/Server.c

```diff
@@ -295,7 +295,31 @@ mp_obj_t common_hal_mdns_server_find(mdns_server_obj_t *self, const char *servic
     return MP_OBJ_FROM_PTR(tuple);
 }
 
-void common_hal_mdns_server_advertise_service(mdns_server_obj_t *self, const char *service_type, const char *protocol, mp_int_t port) {
+static void srv_txt(struct mdns_service *service, void *txt_userdata) {
+    mp_obj_t txt_records = (mp_obj_t)txt_userdata;
+    if (txt_records == mp_const_none) {
+        return;
+    }
+
+    mp_obj_iter_buf_t iter_buf;
+    mp_obj_t iterable = mp_getiter(txt_records, &iter_buf);
+    mp_obj_t txt_record;
+    err_t res;
+    while ((txt_record = mp_iternext(iterable)) != MP_OBJ_STOP_ITERATION) {
+        if (!mp_obj_is_str_or_bytes(txt_record)) {
+            mp_raise_RuntimeError(translate("Failed to add service TXT record; non-string or bytes found in txt_records"));
```

Comment:
This should be a `ValueError`, because it's an argument that's incorrect, and the validation can be done in `shared-bindings`, instead of here, so that the validation code can be used across different ports.

Reply:
`shared-bindings` `wifi/Radio.c` and `i2ctarget/I2CTarget.c` run through an iterable in a shared argument. It's been more common to take a `Sequence`, and then use subscript operations to fetch the values, as in `keypad/Keys.c`. I think I chose `Sequence` for these to ensure that I wasn't passed a never-ending iterable.